OpenSim/Simulation/Model/Model.cpp

```diff
@@ -489,6 +494,46 @@ void Model::createMultibodySystem()
     addToSystem(*_system);
 }
 
+
+void Model::reorderCoordinatesAccordingToSystemMobilities()
+{
+    OPENSIM_THROW_IF_FRMOBJ(!isValidSystem(), Exception,
+        "Attempting to reorder Coordinates according to an invalid System.");
+
+    auto& coordSet = updCoordinateSet();
+
+    // We have a valid MultibodySystem underlying the Coordinates
+    int nc = coordSet.getSize();
+    const SimTK::State& s = getModel().getWorkingState();
+    SimTK_ASSERT_ALWAYS(nc <= s.getNQ(),
+        "Number of Coordinates exceeds the number of mobilities in "
+        "the underlying MultibodySystem.");
```

Comment:
I think that, technically, `getNU()` is the number of mobilities, not `getNQ()`. Perhaps:

"Number of OpenSim Coordinates exceeds the number of generalized coordinates..."